OktaIdxExample/OktaIdxExampleUITests/PhoneLoginScenarioTests.swift

```diff
@@ -46,65 +75,96 @@ final class PhoneLoginScenarioTests: ScenarioTestCase {
+        XCTAssertTrue(factorsPage.continueButton.exists)
+        
+        factorsPage.phoneLabel.tap()
+        
+        XCTAssertTrue(factorsPage.phonePicker.waitForExistence(timeout: .minimal))
+        factorsPage.selectPickerWheel(.sms)
+        
+        
+        XCTAssertTrue(factorsPage.phoneNumberLabel.waitForExistence(timeout: .regular))
+        XCTAssertTrue(factorsPage.phoneNumberField.exists)
+        
+        factorsPage.phoneNumberField.tap()
+        factorsPage.phoneNumberField.typeText("+123456789")
+        factorsPage.continueButton.tap()
+        
+        XCTAssertTrue(app.staticTexts["Unable to initiate factor enrollment: Invalid Phone Number."].waitForExistence(timeout: .regular))
     }
     
-    func testLoginWithInvalidCode() throws {
+    func testLoginWithSMS() throws {
+        try testEnrollWithSMS()
+        
+        app.terminate()
+        app.launch()
+        
         let credentials = try XCTUnwrap(scenario.credentials)
         let signInPage = SignInFormPage(app: app)
         signInPage.signIn(username: credentials.username, password: credentials.password)
-
-        _ = passFactorsEnrollment(phoneNumber: try XCTUnwrap(scenario.profile?.phoneNumber))
+        
+        let factorsPage = FactorsEnrollmentPage(app: app)
+        
+        XCTAssertTrue(factorsPage.phonePicker.waitForExistence(timeout: .minimal))
+        factorsPage.selectPickerWheel(.sms)
+        
+        // Before receiving a code, we must reset all messages.
+        try scenario.resetMessages(.sms)
+        
+        factorsPage.continueButton.tap()
         
         let passcodePage = PasscodeFormPage(app: app)
         XCTAssertTrue(passcodePage.passcodeLabel.waitForExistence(timeout: .regular))
         XCTAssertTrue(passcodePage.passcodeField.exists)
         XCTAssertTrue(passcodePage.resendButton.exists)
         
+        let smsCode = try scenario.receive(code: .sms)
+        
         passcodePage.passcodeField.tap()
-        passcodePage.passcodeField.typeText("12345")
+        passcodePage.passcodeField.typeText(smsCode)
         
         passcodePage.continueButton.tap()
         
-        XCTAssertTrue(app.staticTexts["Invalid code. Try again."].waitForExistence(timeout: .regular))
+        XCTAssertTrue(app.tables.cells["username"].waitForExistence(timeout: .regular))
+        XCTAssertTrue(app.tables.cells["username"].staticTexts[credentials.username].exists)
     }
     
-    private func passFactorsEnrollment(phoneNumber: String) -> Bool {
-        let factorsPage = FactorsEnrollmentPage(app: app)
-        let isUserEnrolled = factorsPage.isUserPhoneEnrolled
-
-        if !isUserEnrolled {
-            XCTAssertTrue(factorsPage.phoneLabel.waitForExistence(timeout: .regular))
-            XCTAssertTrue(factorsPage.continueButton.exists)
-            
-            factorsPage.phoneLabel.tap()
-        }
+    func testLoginWithInvalidCode() throws {
+        try testEnrollWithSMS()
```

Comment:
Perhaps instead of repeating the test to enroll the user, we can use the SDK itself within our scenario setup to enroll the user.  That might make the test more stable, and run faster.  Not necessary for now of course, but something to think about, if you're having problems.

Reply:
Enrolling of user and authentication are slightly different. I ended up with dull code `ifs` when I tried to generalize the code. So in this scenario I think deleting and creating the user between tests, enrolling it before testing specific scenario is a tradeoff. Moreover, order of tests are run random, so we hardly know whether a user is enrolled already or not, so we should it [detect somehow](https://github.com/okta/okta-idx-swift/pull/38/files#diff-70b314b4b0988ed5a5add42c3f3b651396b5390be0fb8129b418e6703dc6d9a2R22) and put [`ifs`](https://github.com/okta/okta-idx-swift/pull/38/files#diff-db750673b977f8f5406987b9269814e327b3ecb76b1e24a25e5f9f78a737c00cR107). 
It makes tests hard to debug.